internal/service/lightsail/container_service.go

```diff
@@ -0,0 +1,997 @@
+package lightsail
+
+import (
+	"context"
+	"log"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/lightsail"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceContainerService() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceContainerServiceCreate,
+		ReadContext:   resourceContainerServiceRead,
+		UpdateContext: resourceContainerServiceUpdate,
+		DeleteContext: resourceContainerServiceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		CustomizeDiff: verify.SetTagsDiff,
+
+		Schema: map[string]*schema.Schema{
+			// required fields
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+			},
+			"power": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"scale": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			// optional fields
+			"deployment": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"container": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"container_name": {
```

Comment:
The AWS Public API documentation for `Container` does not have `container_name`? Please verify.

I do see it defined on the `public_endpoint`, maybe a documentation issue?

https://docs.aws.amazon.com/lightsail/2016-11-28/api-reference/API_Container.html

Reply:
In here https://docs.aws.amazon.com/lightsail/2016-11-28/api-reference/API_CreateContainerServiceDeployment.html, I think `containers` is a "String to [Container](https://docs.aws.amazon.com/lightsail/2016-11-28/api-reference/API_Container.html) object map", where that string is the container's name.